rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -470,7 +476,7 @@ where
                 self.indexer.clone(),
                 Arc::new(self.store.clone()),
                 index_settings.chunk_size,
-                index_settings.from,
+                (index_settings.from as u64).min(u32::MAX as u64) as u32,
```

Comment:
`ForwardBackwardSequenceAwareSyncCursor` should support negative `index_settings.from` values as well.